src/templates.rs

```diff
@@ -30,7 +32,10 @@ pub enum TemplateError {
     TemplateError(#[from] tera::Error),
 }
 
-pub fn get_base_tera(external_context: Option<Context>) -> Result<(Tera, Context), TemplateError> {
+pub fn get_base_tera(
+    external_context: Option<Context>,
+    session: Option<&Session>,
+) -> Result<(Tera, Context), TemplateError> {
     let mut tera = Tera::default();
     let mut context = match external_context {
```

Comment:
```suggestion
    let mut context = external_context.unwrap_or_default();
```
`Context::default()` działa jak `Context::new()` - https://docs.rs/tera/latest/src/tera/context.rs.html#133-135